### ISSUE TITLE
[Bug] "Operation doesn't work on directories." when analyzing APK 

### ISSUE THREAD

**Issue by HuskyHacks** [MobSF/Mobile-Security-Framework-MobSF#2106]
## ENVIRONMENT

```
OS and Version: Ubuntu/REMnux
Python Version: Python 3.10.4
MobSF Version: Docker Latest (Version v3.6.2 Beta)
```

## EXPLANATION OF THE ISSUE

```
Using the official Docker image, MobSF throws an error when analyzing APK files. After uploading an APK, the error is printed to the screen: 
```
![image](https://user-images.githubusercontent.com/57866415/210241180-456489aa-6e01-4eb4-9282-80d378129ccb.png)

## STEPS TO REPRODUCE THE ISSUE

```
1. Pull the latest docker image: $ sudo docker pull opensecurity/mobile-security-framework-mobsf
2. Run the container: $ sudo docker run -it --rm -p 8000:8000 opensecurity/mobile-security-framework-mobsf:latest
3. Create an MSFVenom generated APK or use the one from the Practical Malware Analysis & Triage lab repo (https://github.com/HuskyHacks/PMAT-labs/blob/main/labs/3-6.Mobile-Malware/Android/Malware.android.apk.malz.7z). 7z password is "infected".
4. Rename the APK to test.apk
5. Upload to the file to the MobSF main page.
```

## LOG FILE

```
[INFO] 02/Jan/2023 13:40:00 - Scan Hash: ea7bbcfdd8d77025087b79fde50263c0
[INFO] 02/Jan/2023 13:40:00 - Starting Analysis on: virus.apk
[INFO] 02/Jan/2023 13:40:00 - Generating Hashes
[INFO] 02/Jan/2023 13:40:00 - Unzipping
[INFO] 02/Jan/2023 13:40:00 - APK Extracted
[INFO] 02/Jan/2023 13:40:00 - Getting Hardcoded Certificates/Keystores
[INFO] 02/Jan/2023 13:40:00 - Getting AndroidManifest.xml from APK
[INFO] 02/Jan/2023 13:40:00 - Parsing AndroidManifest.xml
[INFO] 02/Jan/2023 13:40:00 - Fetching icon path
[WARNING] 02/Jan/2023 13:40:00 - androguard cannot find icon resource
[ERROR] 02/Jan/2023 13:40:00 - Error Performing Static Analysis
Traceback (most recent call last):
  File "/home/mobsf/Mobile-Security-Framework-MobSF/mobsf/StaticAnalyzer/views/android/static_analyzer.py", line 182, in static_analyzer
    get_icon_apk(app_dic)
  File "/home/mobsf/Mobile-Security-Framework-MobSF/mobsf/StaticAnalyzer/views/android/icon_analysis.py", line 216, in get_icon_apk
    copy2(src.as_posix(), out.as_posix())
  File "/usr/lib/python3.8/shutil.py", line 435, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.8/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
IsADirectoryError: [Errno 21] Is a directory: '/home/mobsf/.MobSF/uploads/ea7bbcfdd8d77025087b79fde50263c0'
[ERROR] 02/Jan/2023 13:40:00 - [Errno 21] Is a directory: '/home/mobsf/.MobSF/uploads/ea7bbcfdd8d77025087b79fde50263c0'
[ERROR] 02/Jan/2023 13:40:00 - Internal Server Error: /static_analyzer/
```

**Comment by ancodesh** [comment]
I have the same problem

**Comment by thitoo-yf** [comment]
<img width="1437" alt="Screenshot 2023-01-06 at 10 42 07 AM" src="https://user-images.githubusercontent.com/76427440/210928855-5fad701c-a9af-4731-aab5-72b789f522a9.png">
I also found this issue when analyzing the apk.

**Comment by ajinabraham** [comment]
Can any of you share an APK to debug this further?

**Comment by HuskyHacks** [comment]
@ajinabraham The APK in question is an emulated malware sample from my course on malware analysis. It does not actually do anything malicious. It's available in the labs of the course here:
https://github.com/HuskyHacks/PMAT-labs/blob/main/labs/3-6.Mobile-Malware/Android/Malware.android.apk.malz.7z

The file is password protected with the password `infected`. The sample is made with MSFVenom. If you'd like to do a similar test, you can use MSFvenom and follow the instructions from this site: https://resources.infosecinstitute.com/topic/lab-hacking-an-android-device-with-msfvenom/

**Comment by HuskyHacks** [comment]
As a side note, I've confirmed that using the dockerized v3.5.0 build of MobSF does work OK, so something must have changed between then and now.

**Comment by ajinabraham** [comment]
Thanks for the binary. I think, I figured out the issue without it, made a hotfix to possibly fix to address this: https://github.com/MobSF/Mobile-Security-Framework-MobSF/commit/0b259d015ddffd1a96b8972483f735201542dc75 will be available in latest master (3.6.3)

**Comment by HuskyHacks** [comment]
Awesome! Thank you
